pay-api/src/pay_api/services/statement.py

```diff
@@ -392,9 +393,46 @@ def get_summary(auth_account_id: str, statement_id: str = None):
             'total_invoice_due': float(invoices_unpaid_amount) if invoices_unpaid_amount else 0,
             'total_due': total_due,
             'oldest_due_date': oldest_due_date,
-            'short_name_links_count': short_name_links_count
+            'short_name_links_count': short_name_links_count,
+            'is_eft_under_payment': Statement._is_eft_under_payment(auth_account_id, total_due)
+            if calculate_under_payment else None
         }
 
+    @staticmethod
+    def _get_short_name_owing_balance(short_name_id: int) -> Decimal:
+        """Get the total amount owing for a short name statements for all links."""
+        # Pre-filter payment account ids so there is less data to work with
+        accounts_query = (db.session.query(PaymentAccountModel.id)
+                          .join(EFTShortnameLinksModel,
+                                PaymentAccountModel.auth_account_id == EFTShortnameLinksModel.auth_account_id)
+                          .filter(EFTShortnameLinksModel.eft_short_name_id == short_name_id,
+                                  EFTShortnameLinksModel.status_code.in_([EFTShortnameStatus.LINKED.value,
+                                                                          EFTShortnameStatus.PENDING.value])
+                                  ))
+
+        invoices_subquery = (db.session.query(distinct(InvoiceModel.id),
+                                              InvoiceModel.id,
+                                              InvoiceModel.total,
+                                              InvoiceModel.paid)
+                             .join(StatementInvoicesModel, StatementInvoicesModel.invoice_id == InvoiceModel.id)
+                             .filter(and_(InvoiceModel.payment_method_code == PaymentMethod.EFT.value,
+                                          InvoiceModel.invoice_status_code.in_([InvoiceStatus.APPROVED.value,
+                                                                                InvoiceStatus.OVERDUE.value])))
+                             .filter(
+            InvoiceModel.payment_account_id.in_(accounts_query)).subquery())
```

Comment:
Tested against several large accounts for this in clause, at once. Results came back in about 100ms